src/hooks/useLoadClientScript.ts

```diff
@@ -38,44 +38,50 @@ export function useLoadClientScript(props: NostoScriptProps) {
-      const script = createScriptElement(urlPartial)
-      script.onload = scriptOnload
-      document.body.appendChild(script)
-    }
+    if (loadScript) {
+      // Load Nosto client script if not already loaded externally
+      if (!isNostoLoaded() && !shopifyMarkets) {
+        const urlPartial = `/include/${account}`
+        const script = createScriptElement(urlPartial)
+        script.onload = scriptOnload
+        document.body.appendChild(script)
+      }
 
-    // Load Shopify Markets scripts
-    if (shopifyMarkets) {
-      const existingScript = document.querySelector("[nosto-client-script]")
-      const marketId = String(shopifyMarkets?.marketId || "")
-      const language = shopifyMarkets?.language || ""
+      // Load Shopify Markets scripts
+      if (shopifyMarkets) {
+        const existingScript = document.querySelector("[nosto-client-script]")
+        const marketId = String(shopifyMarkets?.marketId || "")
+        const language = shopifyMarkets?.language || ""
 
-      const existingScriptAttributes =
-      existingScript?.getAttribute("nosto-language") !== language ||
-      existingScript?.getAttribute("nosto-market-id") !== marketId
+        const existingScriptAttributes =
+          existingScript?.getAttribute("nosto-language") !== language ||
+          existingScript?.getAttribute("nosto-market-id") !== marketId
 
-      if (!existingScript || existingScriptAttributes) {
-        if (clientScriptLoadedState) {
-          setClientScriptLoadedState(false)
-        }
+        if (!existingScript || existingScriptAttributes) {
+          if (clientScriptLoadedState) {
+            setClientScriptLoadedState(false)
+          }
 
-        const nostoSandbox = document.querySelector("#nosto-sandbox")
+          const nostoSandbox = document.querySelector("#nosto-sandbox")
 
-        existingScript?.parentNode?.removeChild(existingScript)
-        nostoSandbox?.parentNode?.removeChild(nostoSandbox)
+          existingScript?.parentNode?.removeChild(existingScript)
+          nostoSandbox?.parentNode?.removeChild(nostoSandbox)
 
-        const urlPartial =
-        `/script/shopify/market/nosto.js?merchant=${account}&market=${marketId}&locale=${language.toLowerCase()}`
-        const script = createScriptElement(urlPartial)
-        script.setAttribute("nosto-language", language)
-        script.setAttribute("nosto-market-id", marketId)
-        script.onload = scriptOnload
-        document.body.appendChild(script)
+          const urlPartial =
+            `/script/shopify/market/nosto.js?merchant=${account}&market=${marketId}&locale=${language.toLowerCase()}`
+          const script = createScriptElement(urlPartial)
+          script.setAttribute("nosto-language", language)
+          script.setAttribute("nosto-market-id", marketId)
+          script.onload = scriptOnload
+          document.body.appendChild(script)
+        }
       }
+    } else {
+      window.nosto?.reload({
```

Comment:
can you explain what this is for?

Reply:
Currently I am not sure, since Ilja is off and this was added by him previously. I'd simplify it to just leave the state setter toggle the clientScriptLoaded state but left it to the PR since I have no further background to this.